### ISSUE TITLE
Hadolint Override Error Level and Exit Code

### ISSUE THREAD

**Issue by morgan-zylotech** [hadolint/hadolint#673]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
When I downgrade a rule to Info level in the .hadolint.yaml file

```yaml
ignored:
  - DL3005
override:
  info:
    - DL3004
    - DL3008
 ```
I expect that the rule should return an exit code of 0 rather than 1. The rule didn't fail, I just wanted the end user to be aware that there is something potentially wonky. If the general idea of the YAML keys is based off the syslog status codes levels

https://en.wikipedia.org/wiki/Syslog#Severity_level

Anything below a warning should return a valid exit code of 0 due to shell having a very limited set of exit codes to choose from

### Actual behavior

```yaml
override:
  info:
    - DL3004
   style:
    - DL3008
 ```
Both produce exit codes of 1 which is considered a failure (I don't think they should be)

### Steps to reproduce the behavior

```shell
docker run --rm -i -v /your/path/to/hadolint.yaml:/.config/hadolint.yaml hadolint/hadolint < Dockerfile
```

```shell
echo $?
```

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v2.6.0-0-g7160964
```

**Dockerfile** (if relevant)

```Dockerfile
# Use an official Ubuntu Minimal 18.04 runtime as a parent image
FROM ubuntu:bionic AS base
LABEL name="exit status" \
    maintainer="test" \
    version=0.0.1

# Update the base OS
# When you clean up the apt cache by removing /var/lib/apt/lists it
# reduces the image size, since the apt cache is not stored in a layer.
RUN apt-get -y update \
    && apt-get -y upgrade \
    && apt-get install --yes --no-install-recommends software-properties-common
```

**Additional environment details (OS, stack version, etc.)**

**Comment by mvs5465** [comment]
Is downgrading hadolint the only solution to this? Or is there a way to configure this to exit 0 on info?

**Comment by morgan-zylotech** [comment]
I do not know of downgrading the hadolint version would address the problem. This might be a side effect of the limited shell exit codes which is pretty binary as 0 = pass, everything else = fails and how hadolint was implemented

The current workaround to to set the level to ignore which doesn't help.

OR 

force all output to have an exit code of 0 regardless of what happens

```shell
docker run --rm -i -v /your/path/to/hadolint.yaml:/.config/hadolint.yaml hadolint/hadolint < Dockerfile || true
```

OR 



You can set level warnings in other languages. 

This came up as part of a CI/CD pre-commit hook work where the level was downgraded but still failed.

**Comment by lorenzo** [comment]
Have you tried using the `failure-threshold` flag?

**Comment by morgan-zylotech** [comment]
is that a new feature?

```shell
hadolint --help

Usage: hadolint [-v|--version] [--no-fail] [--no-color] [-c|--config FILENAME] 
                [-f|--format ARG] [DOCKERFILE...] [--error RULECODE] 
                [--warning RULECODE] [--info RULECODE] [--style RULECODE] 
                [--ignore RULECODE] 
                [--trusted-registry REGISTRY (e.g. docker.io)] 
                [--require-label LABELSCHEMA (e.g. maintainer:text)] 
                [--strict-labels]
  Lint Dockerfile for errors and best practices
```

**Comment by lorenzo** [comment]
it was introduced a few months ago

**Comment by mvs5465** [comment]
Thanks @lorenzo can confirm that this is working for us now:
```
hadolint Dockerfile --failure-threshold warning
```

**Comment by morgan-zylotech** [comment]
Ok, 

First of all, I want to thank everyone for jumping on this so quickly. Appreciate the effort. I'll close this out. 

I also think the documentation needs to be updated to reflect the changes, I'm happy to do it if needed.

**Comment by lorenzo** [comment]
@morgan-zylotech please do help if you have the time to improve the docs ❤️

**Comment by Pictor13** [comment]
`hadolint Dockerfile --failure-threshold error` in order to still return `exit code 0` when there are warnings.
